### ISSUE TITLE
SuperFluid Donor: Create recurring donation with existing stream balance 

### ISSUE THREAD

**Issue by divine-comedian** [Giveth/giveth-dapps-v2#3359]
Extension of #3091 

**Design is here:**
https://www.figma.com/file/nVoinu0tgJ565enN5R4WDE/GIVeconomy?type=design&node-id=23293-17833&mode=design&t=Is2s27hF4hgjPWca-0

The condition is that as a user I have already made at least 1 recurring donation and have deposited 1 or more tokens into my stream balance(s)

The user has now already an existing stream balance for some tokens and is making a recurring donation to a project 

We show the user this modal when they click `Select Tokens` - this contains a view of a few things:

-  their "Stream Balances" (super token balances)
- How many projects are being funded by the given "Stream balance"
- Estimation of when their stream balance will reach zero (super token balance / net outflow rate per second = seconds left until stream runs out - we can divide this by seconds in a day/week/month to show estimation of when it will run out) 
- A list below of all eligible regular tokens they could use #3002 

![image](https://github.com/Giveth/giveth-dapps-v2/assets/67759413/f43f5157-94fd-43d7-b645-4110e7a758bc)

User selects a token from their existing stream balances they are taken to a view for setting up their stream that contains the following features: 

- [ ] Slider allowing them to choose how much to donate monthly based on their available stream balance & other active recurring donations 
- [ ] link to modal if the user wishes to deposit more tokens  from their wallet into their stream balance `Top-up Stream Balance` #3287 
- [ ] link to `My recurring donations` page #3092 

### a note on estimating time 

- If the users time left on the stream balance is great than one month we always show the estimation in months - rounding to the nearest whole number
- If its less than one month we show it in weeks - rounding to the nearest whole number
- if its less than one week we show it in days - rounding to the nearest whole number
- if its less than one day we show as `< 1 day`

![image](https://github.com/Giveth/giveth-dapps-v2/assets/67759413/a212d71f-df32-439a-a63b-775e0b9f7442)

Super tokens on Optimism for ref.
- DAIx `0x7d342726b69c28d942ad8bfe6ac81b972349d524`
- USDCx `0x8430f084b939208e2eded1584889c9a66b90562f`
- ETHx `0x4ac8bd1bdae47beef2d1c6aa62229509b962aa0d`
- OPx `0x1828bff08bd244f7990eddcd9b19cc654b33cdb4`
- GIVx `0x4caB5B9930210E2EdC6A905B9C75D615872A1A7E`

**Comment by divine-comedian** [comment]
@MohammadPCh - lets make sure we have this one covered

**Comment by MohammadPCh** [comment]
@maryjaf it's ready to test.

**Comment by maryjaf** [comment]
- [ ]  ) 0 month should be shown <0 and "funding 2 projects"
![image](https://github.com/Giveth/giveth-dapps-v2/assets/111529185/dc26ae49-eb21-44ef-83c8-ac7e47e247e0)

**Comment by MohammadPCh** [comment]
@alireza-sharifpour can you take a look?

**Comment by maryjaf** [comment]
> [ ] ) 0 month should be shown <1 and "funding 2 projects"

Thanks, it has been resolved

**Comment by maryjaf** [comment]
- [ ] 2- what should be happen2d by tapping on "manage recurring donation" @divine-comedian @MohammadPCh 
<img width="971" alt="image" src="https://github.com/Giveth/giveth-dapps-v2/assets/111529185/57dc2882-7ab3-4c7b-9de2-d21d44c9e25a">

**Comment by divine-comedian** [comment]
@Tosinolawale - Can you confirm Maryam's comment - we should go to this page, no? My Donations -> Recurring Donations

![image](https://github.com/Giveth/giveth-dapps-v2/assets/67759413/a606020e-0d1d-482f-9e6f-fca134c3e33c)

**Comment by Tosinolawale** [comment]
Yes. Correct @divine-comedian @maryjaf

**Comment by Tosinolawale** [comment]
> * [ ]  2- what should be happen2d by tapping on "manage recurring donation" @divine-comedian @MohammadPCh
> 
> <img alt="image" width="971" src="https://private-user-images.githubusercontent.com/111529185/313456937-57dc2882-7ab3-4c7b-9de2-d21d44c9e25a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA3NjgyMjMsIm5iZiI6MTcxMDc2NzkyMywicGF0aCI6Ii8xMTE1MjkxODUvMzEzNDU2OTM3LTU3ZGMyODgyLTdhYjMtNGM3Yi05ZGUyLWQyMWQ0NGM5ZTI1YS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzE4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMxOFQxMzE4NDNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05NWUwY2IyYTg0ZDE0ODYxMTEyMTEyZWFiNGIzZmQ5M2U0ZDc1OTQ5ZDljYzdlZDIyM2VhMDI4ODEwYTVkNDc1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.iSmkDid9CjeBAEsDt8CGXtYNRq1yc59LE3E_bpDdvSw">

Also, what's with the text box and that formatting? "The make a recurring donation with superfluid explanation looks weird, it was not part of the original design, and if we are sticking with that it could be made better

**Comment by divine-comedian** [comment]
> > * [ ]   2- what should be happen2d by tapping on "manage recurring donation" @divine-comedian @MohammadPCh
> > 
> > <img alt="image" width="971" src="https://private-user-images.githubusercontent.com/111529185/313456937-57dc2882-7ab3-4c7b-9de2-d21d44c9e25a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA3NjgyMjMsIm5iZiI6MTcxMDc2NzkyMywicGF0aCI6Ii8xMTE1MjkxODUvMzEzNDU2OTM3LTU3ZGMyODgyLTdhYjMtNGM3Yi05ZGUyLWQyMWQ0NGM5ZTI1YS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzE4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMxOFQxMzE4NDNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05NWUwY2IyYTg0ZDE0ODYxMTEyMTEyZWFiNGIzZmQ5M2U0ZDc1OTQ5ZDljYzdlZDIyM2VhMDI4ODEwYTVkNDc1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.iSmkDid9CjeBAEsDt8CGXtYNRq1yc59LE3E_bpDdvSw">
> 
> Also, what's with the text box and that formatting? "The make a recurring donation with superfluid explanation looks weird, it was not part of the original design, and if we are sticking with that it could be made better

I added that in, it needed more explanation somewhere or else it wasn't clear for the user what's going on in this page.

**Comment by divine-comedian** [comment]
@MohammadPCh  - this is ready to be updated 

`manage recurring donations` when clicked should open a new tab of the view `my recurring donations`

**Comment by MohammadPCh** [comment]
@maryjaf Item 3 will be addressed once the PR above is merged.

**Comment by maryjaf** [comment]
Thanks 
It has been resolved
